tests/test_uncrustify/failure.py

```diff
@@ -0,0 +1,58 @@
+# Exceptions when a test fails.
+#
+# * @author  Matthew Woehlke    June 2018
+#
+
+
+# =============================================================================
+class Failure(Exception):
+    pass
+
+
+# =============================================================================
+class ExecutionFailure(Failure):
+    # -------------------------------------------------------------------------
+    def __init__(self, exception):
+        self.exception = exception
+
+    # -------------------------------------------------------------------------
+    def __str__(self):
+        return str(self.exception)
+
+
+# =============================================================================
+class MissingFailure(Failure):
+    # -------------------------------------------------------------------------
+    def __init__(self, exception, missing_path):
+        self.exception = exception
```

Comment:
The `self.exception` isn't used, do we need it?

Reply:
Hmm, IIRC the idea was to be able to show information about the test failure. I think that code went missing somewhere...